File: TicketingSystem/ClientApp/src/app/ticket.ts
export class Ticket{
    id: number;
    openedUserId: string;
    title: string;
    resolvedUserId: string;
    resolution: string;
    isOpen: boolean;
    problemDescription: string;
    openDate: Date;
    closeDate: Date;

    constructor(id: number, openedUserID: string, title: string, 
      resolvedUserId: string, resolution: string, isOpen: boolean,
      problemDescription: string, openDate: Date, closeDate: Date) {
            this.id = id;
            this.openedUserId = openedUserID;
            this.title = title;
            this.resolvedUserId = resolvedUserId;
            this.resolution = resolution;
            this.isOpen = isOpen;
            this.problemDescription = problemDescription;
            this.openDate = openDate;
            this.closeDate = closeDate;
        }
}

File: TicketingSystem/Models/Ticket.cs
using System;
using System.Collections.Generic;

namespace TicketingSystem.Models
{
    public partial class Ticket
    {
        public Ticket()
        {
            Favorites = new HashSet<Favorite>();
        }

        public int Id { get; set; }
        public string? OpenedUserId { get; set; }
        public string? Title { get; set; }
        public string? ResolvedUserId { get; set; }
        public string? Resolution { get; set; }
        public bool? IsOpen { get; set; }
        public string? ProblemDescription { get; set; }
        public DateTime OpenDate { get; set; }
        public DateTime? CloseDate { get; set; }

        public virtual ICollection<Favorite> Favorites { get; set; }
    }
}
